docs/fundamentals/code-analysis/quality-rules/ca2018.md

```diff
@@ -0,0 +1,73 @@
+---
+title: "CA2018: `Buffer.BlockCopy` expects the number of bytes to be copied for the `count` argument"
+description: "Learn about code analysis rule CA2018: `Buffer.BlockCopy` expects the number of bytes to be copied for the `count` argument"
+ms.date: 08/17/2021
+ms.topic: reference
+f1_keywords:
+  - "BufferBlockCopyLength"
+  - "CA2018"
+helpviewer_keywords:
+  - "BufferBlockCopyLength"
+  - "CA2018"
+author: mahdiva
+---
+# CA2018: `Buffer.BlockCopy` expects the number of bytes to be copied for the `count` argument
+
+| | Value |
+|-|-|
+| **Rule ID** |CA2018|
+| **Category** |[Reliability](reliability-warnings.md)|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+This rule fires when `Array.Length` is used for the `count` argument on arrays with elements larger than 1 byte.
```

Comment:
This should at least read as `elements of size larger than 1 byte`. Will leave it up to the docs folks on whether we should mention that `byte[], sbyte[] or bool[]` are the only array types where each element is of size 1 byte.